lib/tool-bar-view.coffee

```diff
@@ -61,6 +61,9 @@ module.exports = class ToolBarView extends View
     @.element.addEventListener 'scroll', @drawGutter
     window.addEventListener 'resize', @drawGutter
 
+    if atom.config.get 'tool-bar.automaticallyResizeIcons'
```

Comment:
Can you change this to `atom.config.onDidChange` or `atom.config.observe` (which suites best) so disabling works immediately.